### DISCUSSION TITLE
Is there a way to add new Sonos speakers?

### DISCUSSION THREAD

**Discussion by RetroSven** [avantrec/soco-cli#79] [General]
Is there a way to use soco-cli to add/register a new Sonos speaker outside of the apps?  This would be one more step towards permanently freeing people from the volatile ios/android apps.

**Comment by pwt** [comment]
Hi Sven:

Thanks for your question.

I agree this would be nice, but it's not possible because it's not supported by the SoCo library that underpins SoCo-CLI. And, I doubt we'd be able to add it to SoCo because I think it's one of the functional areas for which encryption is used, and we don't have access to the Sonos encryption keys.

Cheers,
Peter
